Marketplace: table design updates

raicem opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:
Brings design updates to the my subscriptions table. 
- Update table headers and descriptions. The refresh button is now standalone and not a link in the description. 
- Removed the third free-to-install table
- Merges status and product name columns and adds a badge for not connected products
- Add a new element for viewing documentation 

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Build WooCommerce Admin assets by `cd plugins/woocommerce-admin && pnpm run start`
2. Visit the My Subscription page and see the design changes. 
3. Clicking the new **Refresh** button should still work.
4. It isn't likely you won't have expired subscriptions. To make them show up, you may add this snippet into the `WC_Helper::get_subscription_list_data`. This also starts showing the **View Documentation** action dropdown item. You should place some product keys into the arrays defined below. I suggest picking extensions that are installed and not installed. So you see actions in both tables. 

```php
// Make expired. Shows renew button.
$subscriptions_to_expire = [
	'INSERT_PRODUCT_KEY_HERE', // Product keys start with W00. 
];

// Remove product key. Shows subscribe button.
$subscriptions_to_remove_licence = [
	'INSERT_PRODUCT_KEY_HERE', // Product keys start with W00. 
];

foreach ($subscriptions as &$subscription) {
	if ( in_array( $subscription['product_key'], $subscriptions_to_expire, true ) ) {
		$subscription['expired'] = true;
		$subscription['expires'] = 100;
		$subscription['documentation_url'] = 'https://docs.woocommerce.com/document/example/';
	}

	if ( in_array( $subscription['product_key'], $subscriptions_to_remove_licence, true ) ) {
		$subscription['product_key'] = '';
                $subscription['documentation_url'] = 'https://docs.woocommerce.com/document/example/';
	}
}
```

### Screenshot
![image](https://github.com/woocommerce/woocommerce/assets/10389957/95fc2fbb-735b-4135-9576-ec4bb3d60f00)

raicem committed:
Marketplace: table design updates

bgrgicak commented on plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/my-subscriptions.tsx:
```diff
@@ -38,13 +39,24 @@ export default function MySubscriptions(): JSX.Element {
-	const updateConnectionHTML = sprintf(
-		// translators: %s is a link to the update connection page.
+
+	const installedTableDescription = createInterpolateElement(
```
Nice, didn't know you can do that.

raicem replied:
It's pretty powerful. @andfinally showed me this.

bgrgicak commented on plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/my-subscriptions.tsx:
```diff
@@ -38,13 +39,24 @@ export default function MySubscriptions(): JSX.Element {
+					target="_blank"
+					rel="nofollow noopener noreferrer"
+				>
+					your account
```
Should this be translatable?

raicem replied:
Yeah, that confused me too. But I think this text is not used by the `createInterpolateElement` function. The one in the original string is preferred. I confirm this by tweaking the "your account" copy in both places and see which one gets picked up. 

I could have just give a self closing `a` tag to the `createInterpolateElement` and that would work. However, that makes the linter unhappy. 

```js
a: (
	<a
		href="https://woocommerce.com/my-account/my-subscriptions"
		target="_blank"
		rel="nofollow noopener noreferrer"
	/>
),
```

bgrgicak commented on plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/my-subscriptions.tsx:
```diff
@@ -78,44 +90,29 @@ export default function MySubscriptions(): JSX.Element {
+						>
+							<img
+								src={ RefreshIcon }
+								alt="Refresh subscriptions"
```
Translation missing

raicem replied:
Thanks! Added in https://github.com/woocommerce/woocommerce/pull/41066/commits/e8347b13f88677fb480367a22d00750efa97bda5.

bgrgicak commented on plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/rows/actions-dropdown-menu.tsx:
```diff
@@ -2,34 +2,55 @@
+			title: __( 'Manage in Plugins', 'woocommerce' ),
+			icon: <></>,
+			onClick: () => {
+				window.location.href = '/wp-admin/plugins.php';
```
This will break on multisites that use subfolders.
Can we use `wc.wcSettings.ADMIN_URL` to ensure the url is correct?

raicem replied:
`wc.wcSettings.ADMIN_URL` worked but my linter didn't like it. I [used](https://github.com/woocommerce/woocommerce/pull/41066/commits/e8347b13f88677fb480367a22d00750efa97bda5) `import { ADMIN_URL } from '~/utils/admin-settings'` which is used all around WooCommerce Admin.

bgrgicak commented on plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/rows/status-popover.tsx:
```diff
@@ -7,6 +7,7 @@ import { useState } from '@wordpress/element';
 
 export default function StatusPopover( props: {
 	text: string;
+	level: 'error' | 'warning';
```
I see this in a few places. Should we use an enum?

raicem replied:
Yes, makes sense. Implemented in https://github.com/woocommerce/woocommerce/pull/41066/commits/e8347b13f88677fb480367a22d00750efa97bda5.

bgrgicak commented:
I found a few visual things that needs to update.

![Screenshot 2023-11-02 at 08 24 42](https://github.com/woocommerce/woocommerce/assets/1199991/a0df4a82-5c5c-4bc6-bd56-de221f40fc19)

poligilad-auto commented:
> I found a few visual things that needs to update.

@bgrgicak Thanks! Regarding the wrong color I actually I updated the color in the designs, it's better to have the info more legible with the darker gray, plus it's in line with the new table design coming together in core.

raicem committed:
Marketplace: address PR feedback

- Added enum for status levels
- Fixed color for the table rows
- Used ADMIN_URL for referencing plugins page
-

raicem commented:
All updated now, thanks @bgrgicak and @poligilad-auto for checking. 

> Regarding the wrong color I actually I updated the color in the designs, it's better to have the info more legible with the darker gray, plus it's in line with the new table design coming together in core.

I matched the color according to designs. Both were pretty dark but still they weren't the same. We're now on gutenberg gray 900.

bgrgicak reviewed (approved):
Thanks @raicem!

It looks great 🚀 

Please just fix the linter and PHPCS issues before merging.

andfinally commented on plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/my-subscriptions.tsx:
```diff
@@ -39,13 +39,24 @@ export default function MySubscriptions(): JSX.Element {
+	const installedTableDescription = createInterpolateElement(
 		__(
-			'If you don\'t see your subscription, try <a href="%s">updating</a> your connection.',
+			'WooCommerce.com extensions and themes installed on this store. To see all your subscriptions go to <a>your account<custom_icon /></a> on WooCommerce.com.',
```
```suggestion
			'Woo.com extensions and themes installed on this store. To see all your subscriptions go to <a>your account<custom_icon /></a> on Woo.com.',
```

andfinally commented on plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/my-subscriptions.tsx:
```diff
@@ -39,13 +39,24 @@ export default function MySubscriptions(): JSX.Element {
+		{
+			a: (
+				<a
+					href="https://woocommerce.com/my-account/my-subscriptions"
```
```suggestion
					href="https://woo.com/my-account/my-subscriptions"
```

bgrgicak replied:
Could we use the `MARKETPLACE_HOST` constant?

andfinally commented on plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/my-subscriptions.tsx:
```diff
@@ -128,13 +124,13 @@ export default function MySubscriptions(): JSX.Element {
 				<p className="woocommerce-marketplace__my-subscriptions__table-description">
 					{ __(
-						'Your unused Woo.com subscriptions.',
+						"WooCommerce.com subscriptions you haven't used yet.",
```
```suggestion
						"Woo.com subscriptions you haven't used yet.",
```

andfinally commented on plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/my-subscriptions.scss:
```diff
@@ -1,96 +1,110 @@
-	color: var(--wp-red-red-70, #8a2424);
-	background: var(--wp-red-red-0, #fcf0f1);
 	padding: 2px 8px;
+	margin-left: $grid-unit-15; 
```
```suggestion
	margin-left: $grid-unit-15;
	text-align: left;
```

This avoids this on mobile viewports:

<img width="434" alt="image" src="https://github.com/woocommerce/woocommerce/assets/1647564/a9e6f862-0b8d-4ff7-8491-30a11722887f">

We could even add `white-space: nowrap` to get this, but that is more debatable, as the buttons may take up a lot of width in some languages.

<img width="498" alt="image" src="https://github.com/woocommerce/woocommerce/assets/1647564/897417a7-4e71-4737-ada4-b83ad53e548f">

raicem replied:
Happy to accept this but I'd like to make sure I fully understand it:

> This avoids this on mobile viewports:

When we mean "this" what do you mean? Just adding `text-align: left` makes it look like this on my end:
![image](https://github.com/woocommerce/woocommerce/assets/10389957/8a0b1507-30e2-40bb-b176-1ef386623456)

If that's what we're looking for, then I can implement it. 

Agreed that `white-space: nowrap` looks even better.

andfinally replied:
Thanks @raicem, the buttons look fine as is on bigger screens, but on smaller screens the text wraps, and without the `text-align: left`, it looks weird, with the icon off to one side.

<img width="249" alt="image" src="https://github.com/woocommerce/woocommerce/assets/1647564/271b25d7-1ffb-41ec-8cd4-08e3e0530be1">

As an alternative, I'd say let's just add `white-space: nowrap;` - then this situation won't arise.

raicem replied:
Alright, thank you for clarification! Implemented and pushed up now.

andfinally commented on plugins/woocommerce/includes/admin/helper/class-wc-helper.php:
```diff
@@ -1483,23 +1483,24 @@ public static function get_subscription_list_data() {
-				'sites_active'   => 0,
-				'autorenew'      => false,
-				'maxed'          => false,
+				'product_key'       => '',
+				'product_id'        => $data['_product_id'],
+				'product_name'      => $data['Name'],
+				'product_url'       => $data['PluginURI'] ?? '',
+				'zip_slug'          => $data['slug'],
+				'documentation_url' => '',
+				'key_type'          => '',
+				'key_type_label'    => '',
+				'lifetime'          => false,
+				'product_status'    => 'publish',
+				'connections'       => array(),
+				'expires'           => 0,
+				'expired'           => true,
+				'expiring'          => false,
+				'sites_max'         => 0,
+				'sites_active'      => 0,
+				'autorenew'         => false,
+				'maxed'             => false,
```
It might be nice to sort these A-Z, unless we want to keep an order we else elsewhere.

raicem replied:
Hmmmm, I thought it's in order of "importance". I guess this order may not work when reading and editing the code. But when you are looking at this data when debugging, I prefer product name, key, id to be on the top, to help easily identify the item. I wouldn't care about `autorenew` in that context.

bgrgicak replied:
I tried keeping all positions the same between Woo.com, the API response, helper functions, and TS types to make it easier to find things.

Not sure if A-Z is necessary, but anything works as long as it's consistent.

andfinally replied:
Nice, thanks – if we're keeping the order consistent with other places that makes sense.

andfinally commented on plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/rows/functions.tsx:
```diff
@@ -95,17 +103,34 @@ export function productName( subscription: Subscription ): TableRow {
+	const statusElement = statusBadges.map( ( badge, index ) => {
+		return (
+			<StatusPopover
+				key={ index }
```
I'm remembering Michal's warnings about indexes as keys p6JqRr-bar-p2. But I think we're good in this case, as we're not adding or removing items.

andfinally commented on plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/rows/actions-dropdown-menu.tsx:
```diff
@@ -2,34 +2,56 @@
+} ) {
+	const controls = [
+		{
+			title: __( 'Manage in WooCommerce.com', 'woocommerce' ),
```
"In" isn't necessarily wrong, but it's more natural to me to use "on" for a website – like "I saw it on Twitter".

```suggestion
			title: __( 'Manage on Woo.com', 'woocommerce' ),
```

andfinally commented on plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/rows/actions-dropdown-menu.tsx:
```diff
@@ -2,34 +2,56 @@
+			icon: <></>,
+			onClick: () => {
+				window.open(
+					'https://woocommerce.com/my-account/my-subscriptions',
```
```suggestion
					'https://woo.com/my-account/my-subscriptions',
```

andfinally reviewed (approved):
Thanks for all your work on this Cem! 👌 Everything works nicely.

It might be helpful to add a busy state on the Refresh button when clicked, as it takes quite a while for the page to reload. Don't worry about that now though, I'll add a note to the feedback parent issue.

I want to propose this small additional change. When you have extensions in the "Installed on this store" list, but you don't have subscriptions for them, it's weird that we say "Never expires". There's nothing to expire. I reckon we should just show nothing. I also find the comma in the date weird. I also added a note about this to the [feedback issue](18630-gh-Automattic/woocommerce.com)).

```diff
Index: plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/rows/functions.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/rows/functions.tsx b/plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/rows/functions.tsx
--- a/plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/rows/functions.tsx	(revision 0bf89e4467c7d778be624c5c6ec0d07b18a7a23c)
+++ b/plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/rows/functions.tsx	(date 1699375057434)
@@ -143,11 +143,11 @@
 export function expiry( subscription: Subscription ): TableRow {
 	const expiryDate = subscription.expires;
 
-	let expiryDateElement = __( 'Never expires', 'woocommerce' );
+	let expiryDateElement = '';
 
 	if ( expiryDate ) {
 		expiryDateElement = gmdateI18n(
-			'j M, Y',
+			'j M Y',
 			new Date( expiryDate * 1000 )
 		);
 	}
```

raicem committed:
Marketplace: Update WooCommerce.com references to Woo.com

Co-authored-by: And Finally <andfinally@users.noreply.github.com>

raicem committed:
Marketplace: fix how status badges look on narrower screens
